dot/rpc/modules/grandpa.go

```diff
@@ -71,3 +97,101 @@ func (gm *GrandpaModule) ProveFinality(r *http.Request, req *ProveFinalityReques
 
 	return nil
 }
+
+// RoundState returns the state of the current best round state as well as the ongoing background rounds.
+func (gm *GrandpaModule) RoundState(r *http.Request, req *EmptyRequest, res *RoundStateResponse) error {
+	voters := gm.blockFinalityAPI.GetVoters()
+	votersPkBytes := make([]ed25519.PublicKeyBytes, len(voters))
+	for i, v := range voters {
+		votersPkBytes[i] = v.PublicKeyBytes()
+	}
+
+	prevotes := gm.blockFinalityAPI.PreVotes()
+	precommits := gm.blockFinalityAPI.PreCommits()
+
+	missingPrevotes, err := toAddress(difference(votersPkBytes, prevotes))
+	if err != nil {
+		return err
+	}
+
+	missingPrecommits, err := toAddress(difference(votersPkBytes, precommits))
+	if err != nil {
+		return err
+	}
+
+	totalWeight := calcTotalWeight(voters)
+
+	roundstate := RoundStateResponse{
+		SetID: uint32(gm.blockFinalityAPI.GetSetID()),
+		Best: RoundState{
+			Round:           uint32(gm.blockFinalityAPI.GetRound()),
+			ThresholdWeight: calcThresholdWeight(totalWeight),
+			TotalWeight:     totalWeight,
+			Prevotes: Votes{
+				CurrentWeight: calcWeight(voters, prevotes),
+				Missing:       missingPrevotes,
+			},
+			Precommits: Votes{
+				CurrentWeight: calcWeight(voters, precommits),
+				Missing:       missingPrecommits,
+			},
+		},
+		Background: []RoundState{},
+	}
+
+	*res = roundstate
+	return nil
+}
+
+func calcWeight(voters grandpa.Voters, pre map[ed25519.PublicKeyBytes]*grandpa.Vote) uint32 {
+	var weight uint32
+	for pk := range pre {
+		for _, gpv := range voters {
+			if gpv.PublicKeyBytes() == pk {
+				weight += uint32(gpv.ID)
+			}
+		}
+	}
+	return weight
+}
+
+func calcTotalWeight(voters grandpa.Voters) uint32 {
+	var totalWeight uint32
+	for _, v := range voters {
+		totalWeight += uint32(v.ID)
+	}
+
+	return totalWeight
+}
```

Comment:
the `total weight` is currently just the # of authorities at the moment, since all authorities have the same weight

Reply:
just a question: the number of authorities is equal the `len(voters)`?

Reply:
for the current round, yes!